### ISSUE TITLE
nan values in batch normalization during testing phase

### ISSUE THREAD

**Issue by victoryc** [facebookresearch/SparseConvNet#190]
Batch normalization is producing "nan" values for me during testing phase. Any idea what might be the cause and how to fix this?

I do see that BatchNormalization_ForwardPass function takes the boolean parameter "train" and behaves differently during training and testing. But, at the fundamental level, what would be responsible for 'nan's getting generated during testing and how to avoid it?

Thanks.

**Comment by jacknewsom** [comment]
Were you able to resolve this?

**Comment by MarvinLer** [comment]
Ran into this issue, which was due to using a batch size of 1 thus producing nan in running_var
Solved by ensuring a batch size greater than 1
